addons/sale/models/sale.py

```diff
@@ -546,18 +546,19 @@ def _get_tax_amount_by_group(self):
             base_tax = 0
             for tax in line.tax_id:
                 group = tax.tax_group_id
-                res.setdefault(group, 0.0)
+                res.setdefault(group, {'amount':0.0, 'base':0.0})
                 # FORWARD-PORT UP TO SAAS-17
                 price_reduce = line.price_unit * (1.0 - line.discount / 100.0)
                 taxes = tax.compute_all(price_reduce + base_tax, quantity=line.product_uom_qty,
                                          product=line.product_id, partner=self.partner_shipping_id)['taxes']
                 for t in taxes:
-                    res[group] += t['amount']
+                    res[group]['amount'] += t['amount']
+                    res[group]['base'] += t['base']
                 if tax.include_base_amount:
                     base_tax += tax.compute_all(price_reduce + base_tax, quantity=1, product=line.product_id,
                                                 partner=self.partner_shipping_id)['taxes'][0]['amount']
         res = sorted(pycompat.items(res), key=lambda l: l[0].sequence)
-        res = [(l[0].name, l[1]) for l in res]
+        res = [(l[0].name, l[1]['amount'], l[1]['base'], len(res)) for l in res]
```

Comment:
@smetl 
actually, task goal is to display only tax group name + tax amount when invoice/sale order is having only one tax group. And if invoice/sale order is having multiple tax groups then it needs to show tax groups names + base amount + tax amount.
So, in account.invoice we have a tax_line_ids field, so we can use len() like https://github.com/odoo/odoo/pull/16724/commits/e224d0b4fe679e005d215a63b755e65323c973f1#diff-774f4f512fb11d642b30d54aa5980f6bR115

But in sale.order/sale.order.line model we do not have such field, so to find the tax groups count I have to pass the length of tax groups in this function and I checked it in template like https://github.com/odoo/odoo/pull/16724/commits/90ae1dc7bbf5866e5901218bd746532d0fa8aaa5#diff-e7cfd97817c83d4d82652ff1ddcf2d9eR153.